app/controllers/partner_deals_controller.rb

```diff
@@ -15,16 +15,12 @@ def update
     deal = PartnerDeal.find(params[:id])
     # has this user redeemed a code?
     code = deal.codes.where(user: current_user).last
-    if code.nil?
-      # todo: deal with potential race condition
```

Comment:
Once that's in there, consider this LGTM'd.  Looks great, even if I don't think we have a use... yet!